src/lib/sw_engine/tvgSwRenderer.cpp

```diff
@@ -80,7 +80,7 @@ struct SwShapeTask : SwTask
         if (HALF_STROKE(strokeWidth) > 0) {
             sdata->strokeColor(nullptr, nullptr, nullptr, &strokeAlpha);
         }
-        bool validStroke = (strokeAlpha > 0) || sdata->strokeFill();
+        bool validStroke = (static_cast<uint32_t>(strokeAlpha * opacity / 255) > 0) || sdata->strokeFill();
```

Comment:
any necessary for this?

Reply:
above strokeAlpha checks zero case and opacity as well.

Reply:
lets take the shape->opacity(10) and shape->stroke(255, 0, 0, 20)
opacity = 10
alpha = 20
but still, this stroke won't be visible, since 20 * 10 / 255 = 0.8 -> 0
the same is done for a fill l.93-94